### DISCUSSION TITLE
static lib and .pc

### DISCUSSION THREAD

**Discussion by rkd77** [c-ares/c-ares#925] [Q&A] [answered]
Hello,
static lib is libcares_static.a , but in .pc there is -lcares
I want the static library be available out of the box. Currently it is unusable for software which relies on pkgconfig for finding dependencies.

**Comment by bradh352** [comment]
As far as I can tell pkgconfig doesn't support that.  It allows you to specify *additional* libraries when linking statically, but doesn't allow you to change the library name. If using cmake, the cmake modules do allow you to target either-or.  

That said, if you are building c-ares yourself, which if you're linking statically, you probably should ... if you build only the static variant the library name is actually `libcares.a` even when building with cmake.  This is sort of an oddity due to the way cmake works for ADD_LIBRARY() where you can't have 2 libraries with the same name in the same build ... so we had to make "libcares" and "libcares_static".

**Reply by bradh352** [reply]
actually, maybe we can `set_target_properties(cares_static PROPERTIES OUTPUT_NAME cares)` under an `IF (NOT WIN32)` clause as Windows is part of the issue since it uses the same link lib suffix for static and shared ... see : https://stackoverflow.com/a/2152157

Since autotools always generated libcares.so and libcares.a, but cmake generates libcares.so and libcares_static.a making them both behave the same on non-windows platforms might be more proper.

**Comment by rkd77** [comment]
Maybe something like here:
https://alexreinking.com/blog/building-a-dual-shared-and-static-library-with-cmake.html
Build twice, but no conflict in cmake macros.

**Reply by bradh352** [reply, marked as answer]
that looks pretty complex and there could be issues if you need different flags passed to compilation for whatever reason.  Things that come to mind are PIC flags, then of course on Windows there's the symbol export differences for dllimport/dllexport vs static. 
c-ares is still pretty small and compiles fast.  The test cases take many multiples longer to compile so I'm not terribly concerned about double-compilation for static + dynamic builds.

I just created PR #933 for the simple fix.

**Comment by rkd77** [comment]
Thanks.
